code/modules/mining/boulder_processing/brm.dm

```diff
@@ -0,0 +1,142 @@
+#define MANUAL_TELEPORT_SOUND 'sound/machines/mining/manual_teleport.ogg'
+#define AUTO_TELEPORT_SOUND 'sound/machines/mining/auto_teleport.ogg'
+
+/obj/machinery/bouldertech/brm
+	name = "boulder retrieval matrix"
+	desc = "A teleportation matrix used to retrieve boulders excavated by mining NODEs from ore vents."
+	icon_state = "brm"
+	circuit = /obj/item/circuitboard/machine/brm
+	usage_sound = MANUAL_TELEPORT_SOUND
+	processing_flags = START_PROCESSING_MANUALLY
+	boulders_held_max = 2
+	/// Are we trying to actively collect boulders automatically?
+	var/toggled_on = FALSE
+	/// How long does it take to collect a boulder?
+	var/teleportation_time = 1.5 SECONDS
+	/// Cooldown used for left click teleportation.
+	COOLDOWN_DECLARE(manual_teleport_cooldown)
+
+/obj/machinery/bouldertech/brm/Initialize(mapload)
+	. = ..()
+	set_wires(new /datum/wires/brm(src))
+
+/obj/machinery/bouldertech/brm/Destroy()
+	QDEL_NULL(wires)
+	return ..()
+
+/obj/machinery/bouldertech/brm/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, manual_teleport_cooldown))
+		return
+	if(panel_open)
+		balloon_alert(user, "close panel first!")
+		return
+	playsound(src, MANUAL_TELEPORT_SOUND, 50, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	pre_collect_boulder()
+
+	COOLDOWN_START(src, manual_teleport_cooldown, teleportation_time)
+
+/obj/machinery/bouldertech/brm/attackby(obj/item/attacking_item, mob/user, params)
+	if(is_wire_tool(attacking_item) && panel_open)
+		wires.interact(user)
+		return TRUE
+	return ..()
+
+/obj/machinery/bouldertech/brm/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+	if(!anchored)
+		balloon_alert(user, "anchor first!")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	toggle_auto_on(user)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/machinery/bouldertech/brm/process()
+	if(SSore_generation.available_boulders.len < 1)
+		say("No boulders to collect. Entering idle mode.")
+		toggled_on = FALSE
+		update_appearance(UPDATE_ICON_STATE)
+		return PROCESS_KILL
+	for(var/i in 1 to boulders_processing_max)
+		if(pre_collect_boulder())
+			continue
```

Comment:
```suggestion
```

If you do continue the next 3 lines after this don't get executed. How did Byond not complain about this?